src/onnx/parse_split.cpp

```diff
@@ -34,6 +34,131 @@ namespace migraphx {
+                make_op("mul"), chunk_size, info.add_literal(literal{int64_scalar_shape, {n}})),
+            info.add_instruction(make_op("mul"),
+                                 chunk_size,
+                                 info.add_literal(literal{int64_scalar_shape, {n + 1}})));
+    }
+    // last slice: slice(input, starts = {n * chunk_size}); ends = max_int, axes =
+    // {tuned_axis}
+    ret_ins.at(num_outputs - 1) = info.add_instruction(
+        make_op("slice", {{"axes", {tuned_axis}}, {"ends", {std::numeric_limits<int64_t>::max()}}}),
+        args[0],
+        info.add_instruction(make_op("mul"),
+                             chunk_size,
+                             info.add_literal(literal{int64_scalar_shape, {num_outputs - 1}})));
+    return ret_ins;
+}
+
+auto parse_static_split(const onnx_parser::node_info& info,
+                        const onnx_parser& parser,
+                        const std::vector<instruction_ref>& args,
+                        int64_t tuned_axis)
+{
+    const auto& input_shape = args[0]->get_shape();
+    // either static shape or fixed dynamic_dimension for split axis
+    auto tuned_axis_len = input_shape.to_static(0).lens().at(tuned_axis);
+    std::vector<int64_t> vec_splits;
+    if(contains(info.attributes, "split"))
+    {
+        literal s = parser.parse_value(info.attributes.at("split"));
+        s.visit([&](auto v) { vec_splits.assign(v.begin(), v.end()); });
+    }
+    else if(args.size() == 2)
+    {
+        auto s = args[1]->eval();
+        check_arg_empty(s, "PARSE_SPLIT: non-constant `split` input is not supported");
+        s.visit([&](auto v) { vec_splits.assign(v.begin(), v.end()); });
+    }
+    // no split attribute, input is equally divided
+    else
+    {
+        std::size_t num_outputs = info.num_outputs;
+        // the num_outputs attribute seems to be redundant since we already have
+        // node_info::num_outputs, but we can still perform an error check
+        if(contains(info.attributes, "num_outputs"))
+        {
+            num_outputs = parser.parse_value(info.attributes.at("num_outputs")).at<std::size_t>();
+            if(num_outputs != info.num_outputs)
+            {
+                MIGRAPHX_THROW("PARSE_SPLIT: num_outputs attribute " + std::to_string(num_outputs) +
+                               " doesn't match actual number of outputs " +
+                               std::to_string(info.num_outputs) + "!");
+            }
+        }
+        if(tuned_axis_len % num_outputs == 0)
+        {
+            std::size_t chunk_size = tuned_axis_len / num_outputs;
+            vec_splits.resize(num_outputs, chunk_size);
+        }
+        else
+        {
+            std::size_t chunk_size      = tuned_axis_len / num_outputs + 1;
+            std::size_t last_chunk_size = tuned_axis_len - chunk_size * (num_outputs - 1);
+            vec_splits.resize(num_outputs - 1, chunk_size);
+            vec_splits.push_back(last_chunk_size);
```

Comment:
This can potentially lead to resizing and memory allocation twice for the vector. Ideally, you can allocate vector for the splits once for the size (num_outputs) and then put values in that